<launch>
    <arg name="ekf_node_name" default="robot_pose_ekf" />
    <arg name="odom_topic" default="/odom_gps" />
    <arg name="imu_topic" default="/imu/data" />
    <arg name="gps_topic" default="/gps_meas" />
    <arg name="yaw_topic" default="/yaw_odom" />
    <arg name="play_bag" default="true" />
<!--    <arg name="bag_filename" default="/home/yonghui/Datasets/neurobot/localization/2019-10-11/2019-10-11-15-36-36.bag"/>-->
    <arg name="bag_filename" default="/home/yonghui/Datasets/neurobot/localization/2019-10-14/2019-10-14-09-55-50.bag"/>
    <arg name="use_rviz" default="true" />

<!--    <include file="$(find imu_3dm_gx4)/launch/merge.launch" />-->
    <include file="$(find neurobot_data_proprocess)launch/neurobot_odom2gps.launch">
        <arg name="play_bag" value="false" />
        <arg name="use_rviz" value="false" />
        <arg name="use_debug" value="true" />
    </include>

    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="$(arg ekf_node_name)" output="screen" >
<!--        launch-prefix="xterm -e gdb -ex run &#45;&#45;args ">-->
        <param name="base_footprint_frame" value="base_link" />
        <param name="global_frame" value="world" />
        <param name="output_frame" value="odom"/>
        <param name="freq" value="30.0"/>
        <param name="sensor_timeout" value="1.0"/>
        <param name="odom_used" value="true"/>
        <param name="imu_used" value="false"/>
        <param name="vo_used" value="false"/>
        <param name="gps_used" value="true" />
        <param name="yaw_used" value="true" />
        <param name="yaw_split" value="true" />
        <param name="debug" value="false"/>
        <param name="self_diagnose" value="false"/>

        <remap from="odom" to="$(arg odom_topic)" />
        <remap from="imu_data" to="$(arg imu_topic)" />
<!--        <remap from="vo" to="$(arg gps_topic)" />-->
        <remap from="gps" to="$(arg gps_topic)" />
        <remap from="yaw" to="$(arg yaw_topic)" />
    </node>

    <node pkg="neurobot_data_proprocess" type="combine_odom_stamped" name="combine_odom_stamped" output="screen">
        <param name="ekf_node_name" value="$(arg ekf_node_name)" />
        <remap from="gps_meas" to="/gps_meas" />
    </node>

    <group if="$(arg play_bag)">
        <param name="robot_description"
               textfile="$(find neurobot_description)/urdf/neurobot_v2.urdf" />

        <node name="robot_state_publisher" pkg="robot_state_publisher"
              type="robot_state_publisher" />

        <param name="/use_sim_time" value="true" />

        <node name="playbag" pkg="rosbag" type="play"
              args="--clock $(arg bag_filename)" />
    </group>

    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz"
              args="-d $(find robot_pose_ekf)/rviz/neurobot.rviz" />
    </group>

</launch>